products/activedirectory/api.yaml

```diff
@@ -0,0 +1,108 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: ActiveDirectory
+display_name: Managed Microsoft Active Directory
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://managedidentities.googleapis.com/v1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'   
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Domain'
+    kind: 'activedirectory#domain'
+    base_url : projects/{{project}}/locations/global/domains?domainName={{domain_name}}
+    update_verb: :PATCH
+    update_mask: true
+    self_link: '{{name}}'
+    description: Creates a Microsoft AD domain
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Managed Microsoft Active Directory Quickstart': 'https://cloud.google.com/managed-microsoft-ad/docs/quickstarts'
+      api: 'https://cloud.google.com/managed-microsoft-ad/reference/rest/v1/projects.locations.global.domains'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: domainName
+        required: true
+        url_param_only: true
+        description: |
+          The fully qualified domain name. e.g. mydomain.myorganization.com, with the restrictions, 
+          https://cloud.google.com/managed-microsoft-ad/reference/rest/v1/projects.locations.global.domains.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
```

Comment:
Can you add `input: true` to properties that aren't updatable? https://cloud.google.com/managed-microsoft-ad/reference/rest/v1/projects.locations.global.domains/patch lists which ones are.

Reply:
done